notebooks/406-3D-pose-estimation-webcam/406-opencv-3D-pose-estimation.ipynb

```diff
@@ -0,0 +1,587 @@
+    "    \"\"\"\n",
+    "    inference\n",
+    "\n",
+    "    Parameters:\n",
+    "        scaled_img: resized image\n",
+    "        stride: int, the stride of the window\n",
+    "    \"\"\"\n",
+    "    global infer_request, input_tensor_name, model\n",
+    "\n",
+    "    n, c, h, w = model.inputs[0].shape\n",
+    "    # print([n, c, h, w ])\n",
+    "    img = scaled_img[\n",
+    "        0 : scaled_img.shape[0] - (scaled_img.shape[0] % stride),\n",
+    "        0 : scaled_img.shape[1] - (scaled_img.shape[1] % stride),\n",
+    "    ]\n",
+    "\n",
+    "    if h != img.shape[0] or w != img.shape[1]:\n",
+    "        model.reshape(\n",
+    "            {input_tensor_name: PartialShape([n, c, img.shape[0], img.shape[1]])}\n",
+    "        )\n",
+    "        compiled_model = ie_core.compile_model(model, \"GPU\")\n",
+    "        infer_request = compiled_model.create_infer_request()\n",
+    "\n",
+    "    img = np.transpose(img, (2, 0, 1))[\n",
+    "        None,\n",
+    "    ]\n",
+    "    infer_request.infer({input_tensor_name: img})\n",
+    "    results = {\n",
+    "        name: infer_request.get_tensor(name).data[:]\n",
+    "        for name in {\"features\", \"heatmaps\", \"pafs\"}\n",
+    "    }\n",
+    "\n",
+    "    results = (results[\"features\"][0], results[\"heatmaps\"][0], results[\"pafs\"][0])\n",
+    "\n",
+    "    return results\n",
+    "\n",
+    "\n",
+    "def rotate_poses(poses_3d, R, t):\n",
+    "    \"\"\"\n",
+    "    Rotating human coordinates\n",
+    "\n",
+    "    Parameters:\n",
+    "        poses_3d: The position of the human body in the 3-D coordinate system\n",
+    "        R: int, the rotation matrix\n",
+    "        t: int, the translation matrix\n",
+    "    \"\"\"\n",
+    "    R_inv = np.linalg.inv(R)\n",
+    "    for pose_id in range(poses_3d.shape[0]):\n",
+    "        pose_3d = poses_3d[pose_id].reshape((-1, 4)).transpose()\n",
+    "        pose_3d[0:3] = np.dot(R_inv, pose_3d[0:3] - t)\n",
+    "        poses_3d[pose_id] = pose_3d.transpose().reshape(-1)\n",
+    "\n",
+    "    return poses_3d"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14d3e4b2-34b5-4a91-9608-df17c50b971d",
+   "metadata": {},
+   "source": [
+    "### Draw 2D Pose Overlays\n",
+    "We need to define some connections between joints in advance, so that we can draw the structure of the human body in the image when we get the results.\n",
+    "Joints are drawn as circles and limbs are drawn as lines. The code is based on the [3D Human Pose Estimation Demo](https://github.com/openvinotoolkit/open_model_zoo/tree/master/demos/human_pose_estimation_3d_demo/python) from Open Model Zoo."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "22fd3e08-ed3b-44ac-bd07-4a80130d6681",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # 3D edge index array\n",
+    "body_edges = np.array(\n",
+    "    [\n",
+    "        [11, 10],\n",
```

Comment:
@Could you add some comments to explain the usage of these body edges ?  what do they stand for ?